bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/device/feature/CommandHandler.java

```diff
@@ -0,0 +1,2350 @@
+            double increment = SIUnits.CELSIUS.equals(unit) ? 0.5 : 1;
+            return Math.round(value / increment) * increment; // round in increment based on temperature unit
+        }
+
+        private Unit<Temperature> getTemperatureUnit() {
+            String scale = getParameterAsString("scale", "");
+            switch (scale) {
+                case "celsius":
+                    return SIUnits.CELSIUS;
+                case "fahrenheit":
+                    return ImperialUnits.FAHRENHEIT;
+                default:
+                    logger.debug("{}: no valid temperature scale parameter found, defaulting to: CELSIUS", nm());
+                    return SIUnits.CELSIUS;
+            }
+        }
+    }
+
+    /**
+     * Custom time quantity type command handler based of parameters
+     */
+    public static class CustomTimeCommandHandler extends CustomCommandHandler {
+        CustomTimeCommandHandler(DeviceFeature feature) {
+            super(feature);
+        }
+
+        @Override
+        public boolean canHandle(Command cmd) {
+            return cmd instanceof QuantityType;
+        }
+
+        @Override
+        protected double getValue(Command cmd) {
+            @SuppressWarnings("unchecked")
+            QuantityType<Time> time = (QuantityType<Time>) cmd;
+            Unit<Time> unit = getTimeUnit();
+            return Objects.requireNonNullElse(time.toInvertibleUnit(unit), time).doubleValue();
+        }
+
+        private Unit<Time> getTimeUnit() {
+            String scale = getParameterAsString("scale", "");
+            switch (scale) {
+                case "hour":
+                    return Units.HOUR;
+                case "minute":
+                    return Units.MINUTE;
+                case "second":
+                    return Units.SECOND;
+                default:
+                    logger.debug("{}: no valid time scale parameter found, defaulting to: SECONDS", nm());
+                    return Units.SECOND;
+            }
+        }
+    }
+
+    /**
+     * Generic on/off abstract command handler
+     */
+    public abstract static class OnOffCommandHandler extends CommandHandler {
+        OnOffCommandHandler(DeviceFeature feature) {
+            super(feature);
+        }
+
+        @Override
+        public boolean canHandle(Command cmd) {
+            return cmd instanceof OnOffType;
+        }
+
+        @Override
+        public void handleCommand(InsteonChannelConfiguration config, Command cmd) {
+            try {
+                int cmd1 = getCommandCode(config, cmd);
+                int level = getLevel(config, cmd);
+                int group = getGroup(config);
+                // ignore request if cmd1/level not defined, send broadcast msg if group defined, otherwise direct msg
+                if (cmd1 == -1 || level == -1) {
+                    if (logger.isDebugEnabled()) {
```

Comment:
Checking `if (logger.isDebugEnabled()) {` is only needed for  expensive operations, getting the class name is not expensive.  There are more of these checks in this area. I tis minor, but you might want to adjust it.

Reply:
Done.